### PR TITLE
Add mixed precision doc

### PR DESCRIPTION
**Pull request by rohan-varma**
Adds docstring for mixed precision

### CODE REVIEW

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -320,6 +320,13 @@ class FullyShardedDataParallel(nn.Module):
             the near future. It allows users to enable two different backward_prefetch
             algorithms to help backward communication and computation overlapping.
             Pros and cons of each algorithm is explained in the class ``BackwardPrefetch``.
+        mixed_precision: (Optional[MixedPrecision]): A ``MixedPrecision`` instance
+            describing the mixed precision training config to be used. ``MixedPrecision``
+            supports configuring paramter, buffer, and gradient communication dtype. Note
```
Comment (kumpera) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:325]:
```suggestion
            supports configuring parameter, buffer, and gradient communication dtype. Note
```

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 2]
```diff
@@ -320,6 +320,13 @@ class FullyShardedDataParallel(nn.Module):
...
+            that only floating point data is cast to the reduced precision. This allows
+            users potential memory saving and training speedup while trading off
```
Comment (kumpera) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:327]: Do we have reference on how to understand the tradeoff here, in particular the reduces precision?

**Review by kumpera — CHANGES_REQUESTED** [review]
typo.

File: torch/distributed/fsdp/fully_sharded_data_parallel.py (hunk 2 above)
Comment (rohan-varma) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:327]: It will vary based on workload, hw type, and mixed precision config (how parameters, buffers, and gradient communication can be configured differently). Currently other than suggesting bfloat16 as a first step if the HW supports it otherwise fp16 + ShardedGradScaler, it will rely on user experimentation and testing which could possibly be improved by autotuning framework in the future.

**Review by awgu — APPROVED** [review]
Nice!
